src/maybe.ts

```diff
@@ -15,6 +15,17 @@ export enum Variant {
   Nothing = 'Nothing',
 }
 
+export interface JustJsonShape {
```

Comment:
Let's make this `JustJSON<T>`, with `value: T`. We *can* safely make that move, because we always know what `T` is as handed back. Whether the underlying types are implemented as something are *themselves* properly represented by e.g `JSON.stringify` is not this library's responsibility, but the responsibility of whoever owns *that type*. So:

```suggestion
export interface JustJSON<T> {
```